authenticate/handlers.go

```diff
@@ -84,14 +84,14 @@ func (a *Authenticate) mountDashboard(r *mux.Router) {
 		AllowedHeaders:   []string{"*"},
 	})
 	sr.Use(c.Handler)
+	sr.Use(a.RetrieveSession)
```

Comment:
I think `RetrieveSession` triggers authentication? In which case the `signed_out` page wouldn't work properly?

Reply:
Ok maybe it's just the `VerifySession` that does that.

Reply:
Yes, I think it's `VerifySession` that will trigger authentication. I think `RetrieveSession` will only populate some values into the request context: https://github.com/pomerium/pomerium/blob/a771b82a7246ccab3d40dc0fe9d49ce683367ca1/internal/sessions/middleware.go#L25-L28

Let me see if I can add a test case to verify this behavior.

Reply:
OK, I've added a new test case in [authenticate/handlers_test.go](https://github.com/pomerium/pomerium/pull/4765/files#diff-05242e92555eb106ddb7a54833cc89eacb25138dc4f36d52d983b48f95168683).